std/array.d

```diff
@@ -3339,6 +3340,69 @@ if (isDynamicArray!A)
     app.put(r[]);
 }
 
+unittest // Issue 13300
+{ 
+    // Issue 13300: appender must work with non pure postblit
+    import std.array : array;
+
+    struct Simple13300
+    {
+        @disable this(); 
+        this(this) { import std.stdio : stdout; stdout; }  // Not pure!
+
+        private:
+        this(int tmp) { }
```

Comment:
What is the purpose of disabling `this()` and declaring a constructor? At first glance, it looks to me like they're completely unnecessary and that simply using the `init` value in the range would be fine.

Reply:
See comment below